homeassistant/components/mqtt/discovery.py

```diff
@@ -62,6 +142,31 @@
             return
 
         payload = dict(payload)
+
+        for key in list(payload.keys()):
+            abbreviated_key = key
+            # Pattern to match one ore or word characters, excluding _, and:
+            # - at beginning of string or prefixed by _
+            # - at end of string or suffixed by _
+            pattern = r'(?:(?<=^)|(?<=_))[^\W_]+(?=_|$)'
+
+            def expand(matchobj):
+                """Expand a possibly abbreviated word."""
+                abbreviation = matchobj.group(0)
+                if abbreviation in ABBREVIATIONS:
+                    return ABBREVIATIONS[abbreviation]
+                return abbreviation
+
+            key = re.sub(pattern, expand, key)
+            payload[key] = payload.pop(abbreviated_key)
+
+        if TOPIC_PREFIX in payload:
+            prefix = payload[TOPIC_PREFIX]
+            for key, value in payload.items():
+                if value.startswith(TOPIC_PREFIX) and key.endswith('_topic'):
```

Comment:
`value[0]` will be faster.

Reply:
Alright, but then this will only work if TOPIC_PREFIX is a single character long.
Maybe that's a reasonable limitation?